xc-tool/src/main.rs

```diff
@@ -0,0 +1,72 @@
+mod cli;
+mod util;
+
+use clap::Parser;
+
+use std::io::{self, Write};
+
+use cli::{Action, Cli, CliInteractive};
+
+use util::ErrBox;
+
+pub const PROMPT: &'static str = concat!(env!("CARGO_PKG_NAME"), "> ");
+
+fn main() -> Result<(), ErrBox> {
+    let cli = Cli::parse();
+
+    // Handle interactive mode separately
+    if cli.action == Action::Interactive {
+        start_cli_interactive(cli)?;
+    } else {
+        handle_action_noninteractive(&cli)?;
+    }
+    Ok(())
```

Comment:
It might be best to keep the interactive stuff stashed in a separate branch, and and merge it in later if we decide to go with Rust for the tool. The code should initially be a very simple sanity-check of using the available Rust SDKs.